plugins/instanceof-checkcast/src/main/java/org/qbicc/plugin/instanceofcheckcast/ClassInitializerRegister.java

```diff
@@ -0,0 +1,54 @@
+package org.qbicc.plugin.instanceofcheckcast;
+
+import java.util.function.Consumer;
+
+import org.qbicc.context.ClassContext;
+import org.qbicc.context.CompilationContext;
+import org.qbicc.plugin.reachability.RTAInfo;
+import org.qbicc.type.definition.DefinedTypeDefinition;
+import org.qbicc.type.definition.LoadedTypeDefinition;
+import org.qbicc.type.definition.element.InitializerElement;
+
+/**
+ * Register all reachable class's class initalizer methods as entrypoints
+ * so they survive to be lowered into functions.
+ * This is required as we lose info after ReachabilityBlockBuilder required
+ * to otherwise keep them alive.
+ * 
+ * Eventually, this will need to skip an class initializer that was run as
+ * part of the build process.
+ */
+public class ClassInitializerRegister implements Consumer<CompilationContext>  {
+
+    @Override
+    public void accept(CompilationContext ctxt) {
+        RTAInfo rtaInfo = RTAInfo.get(ctxt);
+		
+        ClassContext classContext = ctxt.getBootstrapClassContext();
+        DefinedTypeDefinition jloDef = classContext.findDefinedType("java/lang/Object");
+        LoadedTypeDefinition jlo = jloDef.load();
+
+        // Code below uses #hasMethodBody() to filter out the empty
+        // class initializers and to avoid processing internal arrays
+
+        // Register Object's <clinit> if it exists
+        InitializerElement object_clinit = jlo.getInitializer();
+        if (object_clinit != null && object_clinit.hasMethodBody()) {
+            ctxt.registerEntryPoint(object_clinit);
```

Comment:
See https://github.com/qbicc/qbicc/pull/501#discussion_r636994590 for original discussion on registering entrypoints

Reply:
The point of that discussion got lost but I'd rather see the reachable initializers be re-enqueued at every stage.  Really what I'd like to do eventually is to completely lock out the `registerEntryPoint` method once compilation begins, to force plugins to both know and control the reachable set at every stage, but we have a few cases where this would be problematic at present.